contracts/OrderMixin.sol

```diff
@@ -141,60 +146,92 @@ abstract contract OrderMixin is
     /// @notice Fills an order. If one doesn't exist (first fill) it will be created using order.makerAssetData
     /// @param order Order quote to fill
     /// @param signature Signature to confirm quote ownership
-    /// @param makingAmount Making amount
-    /// @param takingAmount Taking amount
-    /// @param thresholdAmount Specifies maximum allowed takingAmount when takingAmount is zero, otherwise specifies minimum allowed makingAmount
+    /// @param amounts [Making amount, Taking amount, Threshold amount]
     function fillOrder(
         Order memory order,
         bytes calldata signature,
-        uint256 makingAmount,
-        uint256 takingAmount,
-        uint256 thresholdAmount
+        uint256[3] memory amounts
```

Comment:
I think it's easier to define a `struct` for this `OrderAmounts` vs using array

A struct should also solve the too many variable problem because the struct itself only count as 1 stack var.